File: 백준/Silver/20950. 미술가 미미/미술가 미미.py
import sys

input = sys.stdin.readline

INF = float("inf")


def solve(curr, count, r, g, b):
    global answer

    if count == 7 or curr == n:
        if count < 2:
            return

        r //= count
        g //= count
        b //= count

        answer = min(answer, abs(r - gomduri[0]) + abs(g - gomduri[1]) + abs(b - gomduri[2]))
        return

    solve(curr + 1, count + 1, r + colors[curr][0], g + colors[curr][1], b + colors[curr][2])
    solve(curr + 1, count, r, g, b)


n = int(input())
colors = [list(map(int, input().split())) for _ in range(n)]
gomduri = list(map(int, input().split()))

isSelected = [False] * n
answer = INF

solve(0, 0, 0, 0, 0)

print(answer)

File: 백준/Silver/20950. 미술가 미미/미술가 미미.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {

  static StringBuilder sb = new StringBuilder();
  static int n, answer;
  static int[][] colors;
  static int[] targetColor;

  public static void main(String[] args) throws IOException {
    setUp();

    answer = Integer.MAX_VALUE;
    solve(0, 0, 0, 0, 0);

    sb.append(answer);
    output();
  }

  private static void solve(int curr, int r, int g, int b, int count) {
    if (7 < count) {
      return;
    }
    if (curr == n) {
      if (1 < count) {
        int diff = Math.abs(targetColor[0] - r / count) + Math.abs(targetColor[1] - g / count) + Math.abs(targetColor[2] - b / count);
        answer = Math.min(answer, diff);
      }
      return;
    }

    solve(curr + 1, r + colors[curr][0], g + colors[curr][1], b + colors[curr][2], count + 1);
    solve(curr + 1, r, g, b, count);

  }


  private static void setUp() throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st;

    n = Integer.parseInt(br.readLine());
    colors = new int[n][3];
    for (int i = 0; i < n; i++) {
      st = new StringTokenizer(br.readLine());
      for (int j = 0; j < 3; j++) {
        colors[i][j] = Integer.parseInt(st.nextToken());
      }
    }

    targetColor = new int[3];
    st = new StringTokenizer(br.readLine());
    for (int i = 0; i < 3; i++) {
      targetColor[i] = Integer.parseInt(st.nextToken());
    }
  }

  private static void output() throws IOException {
    BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    bw.write(sb.toString());
    bw.flush();
    bw.close();
  }

}